frontend/src/hooks/use-visible/use-visible.hook.ts

```diff
@@ -0,0 +1,30 @@
+import * as React from 'react';
+
+type VisibleHookType = {
+  ref: React.RefObject<HTMLDivElement>;
+  isVisible: boolean;
+  setIsVisible: React.Dispatch<React.SetStateAction<boolean>>;
+};
+
+const useVisible = (isInitialIsVisible: boolean): VisibleHookType => {
+  const [isVisible, setIsVisible] = React.useState<boolean>(isInitialIsVisible);
+  const ref = React.useRef<HTMLDivElement>(null);
+
+  const handleClickOutside = (event: MouseEvent) => {
+    const element = event.target as HTMLElement;
```

Comment:
```suggestion
  const handleClickOutside = (evt: ButtonClickEvent) => {
    const element = evt.target as HTMLElement;
```
Never use `event` for the event param name

We already have `ButtonClickEvent` type in the common folder. Check it